tool_path_planner/src/halfedge_boundary_finder.cpp

```diff
@@ -0,0 +1,327 @@
+/**
+ * @author Jorge Nicho
+ * @file mesh_boundary_finder.cpp
+ * @date Dec 5, 2019
+ * @copyright Copyright (c) 2019, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <boost/bimap.hpp>
+#include <boost/make_shared.hpp>
+#include <pcl/geometry/triangle_mesh.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl/PointIndices.h>
+#include <pcl/kdtree/kdtree_flann.h>
+#include <console_bridge/console.h>
+#include <noether_conversions/noether_conversions.h>
+#include <tool_path_planner/half_edge_boundary_finder.h>
+#include <tool_path_planner/utilities.h>
+
+
+using MeshTraits = pcl::geometry::DefaultMeshTraits<std::size_t>;
+typedef pcl::geometry::TriangleMesh< MeshTraits > TraingleMesh;
+
+boost::optional<TraingleMesh> createTriangleMesh(const pcl::PolygonMesh& input_mesh)
+{
+  std::stringstream ss;
+  TraingleMesh mesh;
+  typedef boost::bimap<uint32_t, TraingleMesh::VertexIndex> MeshIndexMap;
+  MeshIndexMap mesh_index_map;
+  for (size_t ii = 0; ii < input_mesh.polygons.size(); ++ii)
```

Comment:
Could use range for

Reply:
This was extracted from the deepest darkest corners of godel but yes I'll make the update